solr/solrj/src/test/org/apache/solr/client/solrj/response/TermsResponseTest.java

```diff
@@ -31,16 +35,19 @@ public class TermsResponseTest extends EmbeddedSolrServerTestBase {
 
   @BeforeClass
   public static void beforeClass() throws Exception {
-    initCore();
+    solrClientTestRule.startSolr(LuceneTestCase.createTempDir("solrhome"));
```

Comment:
Does this need the "solrhome" parameter?   When I see it, it makes me think it's important somehow..   I saw we also have the use of `.createTempDir()` with no name, is that better in the sense of conveying that it isn't important to have a named directory?

Reply:
> Does this need the "solrhome" parameter? When I see it, it makes me think it's important somehow

I suppose you are wondering, could we have a default of some temp directory so that the test doesn't even need to specify it?  We could do that.  I leaned to specifying it because this is such a foundational thing to Solr.

>  I saw we also have the use of .createTempDir() with no name

It's cosmetic so that if we debug issues, it's more evident what purpose this temp dir is for.